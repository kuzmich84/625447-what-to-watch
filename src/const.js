export const tabsTitle = [`Overview`, `Details`, `Reviews`];

export const AuthorisationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};
