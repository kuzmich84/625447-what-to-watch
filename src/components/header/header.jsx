import React from "react";
import {getAuthorisationStatus, getAvavtar} from "../../store/selectors";
import {connect} from "react-redux";
import {AuthorisationStatus} from "../../const";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";

const Header = ({authorisationStatus, avatar}) => {

  return (
    <header className="page-header movie-card__head">
      <div className="logo">
        <a className="logo__link">
          <span className="logo__letter logo__letter--1">W</span>
          <span className="logo__letter logo__letter--2">T</span>
          <span className="logo__letter logo__letter--3">W</span>
        </a>
      </div>

      <div className="user-block">
        {authorisationStatus === AuthorisationStatus.AUTH
          ? <div className="user-block__avatar">
            <img src={avatar} alt="User avatar" width="63" height="63"/>
          </div>
          : <Link to="/login" className="page-title">Sign in</Link>
        }

      </div>
    </header>
  );
};

const mapStateToProps = (state) => ({
  authorisationStatus: getAuthorisationStatus(state),
  avatar: getAvavtar(state),
});

export default connect(mapStateToProps)(Header);

Header.propTypes = {
  authorisationStatus: PropTypes.string.isRequired,
  avatar: PropTypes.string,
};
