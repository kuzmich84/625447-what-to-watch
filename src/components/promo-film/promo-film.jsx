import React from "react";
import Header from "../header/header";
import {getAuthorisationStatus, getPromoFilm} from "../../store/selectors"
import {connect} from "react-redux";
import {showVideoPage} from "../../store/action";
import PropTypes from "prop-types";
import AddMyListContainer from "../add-my-list/add-my-list-container";
import {postFavorite} from "../../store/api-actions";
import {AuthorisationStatus} from "../../const"

const PromoFilm = ({promo, showVideoPageAction, postFilmFavorite, authorizationStatus}) => {
  const {name, genre, released, backgroundImage, posterImage, isFavorite, id} = promo;

  function handlerShowVideoPage() {
    showVideoPageAction(true);
  }

  return (
    <section className="movie-card">
      <div className="movie-card__bg">
        <img src={backgroundImage} alt={name}/>
      </div>

      <h1 className="visually-hidden">WTW</h1>

      <Header page={`main`}/>

      <div className="movie-card__wrap">
        <div className="movie-card__info">
          <div className="movie-card__poster">
            <img src={posterImage} alt={name}
              width="218" height="327"/>
          </div>

          <div className="movie-card__desc">
            <h2 className="movie-card__title">{name}</h2>
            <p className="movie-card__meta">
              <span className="movie-card__genre">{genre}</span>
              <span className="movie-card__year">{released}</span>
            </p>

            <div className="movie-card__buttons">
              <button className="btn btn--play movie-card__button" type="button" onClick={handlerShowVideoPage}>
                <svg viewBox="0 0 19 19" width="19" height="19">
                  <use xlinkHref="#play-s"/>
                </svg>
                <span>Play</span>
              </button>
              {authorizationStatus === AuthorisationStatus.AUTH && <AddMyListContainer filmId={id} isFavorite={isFavorite} postFilmFavorite={postFilmFavorite}/> }
            </div>
          </div>
        </div>
      </div>
    </section>
  );
};

const mapStateToProps = (state) => ({
  promo: getPromoFilm(state),
  authorizationStatus: getAuthorisationStatus(state),
});

const mapDispatchToProps = (dispatch) => ({
  showVideoPageAction(value) {
    dispatch(showVideoPage(value));
  },
  postFilmFavorite(filmId, status) {
    dispatch(postFavorite(filmId, status));
  }
});
export {PromoFilm};
export default connect(mapStateToProps, mapDispatchToProps)(PromoFilm);

PromoFilm.propTypes = {
  promo: PropTypes.object.isRequired,
  showVideoPageAction: PropTypes.func.isRequired,
  postFilmFavorite: PropTypes.func.isRequired,
};
