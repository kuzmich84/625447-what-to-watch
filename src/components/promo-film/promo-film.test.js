import React from "react";
import renderer from "react-test-renderer";
import {PromoFilm} from "./promo-film";
import mockFilmList from "../../mocks/films";
import {Header} from "../header/header";
import {AuthorisationStatus} from "../../const";

const noop = () => {
};

test(`Should PromoFilm component render correctly`, () => {
  const tree = renderer
    .create(
        <PromoFilm
          showVideoPageAction={noop}
          postFilmFavorite={noop}
          promo={mockFilmList[3]}
        >
          <React.Fragment/>
        </PromoFilm>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
