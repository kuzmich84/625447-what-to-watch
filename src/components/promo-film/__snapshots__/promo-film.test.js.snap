// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should PromoFilm component render correctly 1`] = `
<section
  className="movie-card"
>
  <div
    className="movie-card__bg"
  >
    <img
      alt="Orlando"
    />
  </div>
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>
  <div
    className="movie-card__wrap"
  >
    <div
      className="movie-card__info"
    >
      <div
        className="movie-card__poster"
      >
        <img
          alt="Orlando"
          height="327"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          Orlando
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            Drama
          </span>
          <span
            className="movie-card__year"
          >
            1992
          </span>
        </p>
        <div
          className="movie-card__buttons"
        >
          <button
            className="btn btn--play movie-card__button"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="19"
              viewBox="0 0 19 19"
              width="19"
            >
              <use
                xlinkHref="#play-s"
              />
            </svg>
            <span>
              Play
            </span>
          </button>
          <button
            className="btn btn--list movie-card__button"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="20"
              viewBox="0 0 19 20"
              width="19"
            >
              <use
                xlinkHref="#add"
              />
            </svg>
            <span>
              My list
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</section>
`;
